sdk/src/azure/core/az_http_policy.c

```diff
@@ -42,21 +42,44 @@ AZ_NODISCARD az_result az_http_pipeline_policy_apiversion(
   return _az_http_pipeline_nextpolicy(ref_policies, ref_request, ref_response);
 }
 
+#define _az_TELEMETRY_ID_PREFIX "azsdk-c-"
+#define _az_TELEMETRY_ID_PREFIX_LENGTH (sizeof(_az_TELEMETRY_ID_PREFIX) - 1)
+
 AZ_NODISCARD az_result az_http_pipeline_policy_telemetry(
     _az_http_policy* ref_policies,
     void* ref_options,
     az_http_request* ref_request,
     az_http_response* ref_response)
 {
+  _az_PRECONDITION_NOT_NULL(ref_options);
+
+  // Format spec: https://azure.github.io/azure-sdk/general_azurecore.html#telemetry-policy
+  uint8_t telemetry_id_buffer[200] = _az_TELEMETRY_ID_PREFIX;
+  az_span telemetry_id = AZ_SPAN_FROM_BUFFER(telemetry_id_buffer);
+  {
+    az_span remainder = az_span_slice_to_end(telemetry_id, _az_TELEMETRY_ID_PREFIX_LENGTH);
 
-  _az_http_policy_telemetry_options* options = (_az_http_policy_telemetry_options*)(ref_options);
+    _az_http_policy_telemetry_options* options = (_az_http_policy_telemetry_options*)(ref_options);
+    az_span const component_name = options->component_name;
+    _az_PRECONDITION_VALID_SPAN(component_name, 1, false);
+    remainder = az_span_copy(remainder, component_name);
+
+    remainder = az_span_copy_u8(remainder, '/');
+    remainder = az_span_copy(remainder, AZ_SPAN_FROM_STR(AZ_SDK_VERSION_STRING));
+
+    telemetry_id = az_span_slice(
+        telemetry_id, 0, (int32_t)(az_span_ptr(remainder) - az_span_ptr(telemetry_id)));
```

Comment:
What is this code doing exactly? Can you use `_az_span_diff`? 

Deserves a comment.